### PR TITLE
TST: Add unit test for out=None of np.einsum

### PR DESCRIPTION
**Pull request by sgasse**
Ensure that explicitly stating out=None does not raise an error in
np.einsum, see #15776 and #15256

### CODE REVIEW

File: numpy/core/tests/test_einsum.py [hunk 1]
```diff
@@ -605,6 +605,11 @@ def test_einsum_misc(self):
                      [[[1,  3], [3,  9], [5, 15], [7, 21]],
                      [[8, 16], [16, 32], [24, 48], [32, 64]]])
 
+        # Ensure explicitly setting out=None does not cause an error
+        # see issue #15776 and issue #15256
+        assert_equal(np.einsum('i,j', [1], [2], out=None),
+                               [[2]])
```
Comment (eric-wieser) [inline numpy/core/tests/test_einsum.py]: nit: Indents are misleading here:
```suggestion
        assert_equal(np.einsum('i,j', [1], [2], out=None), [[2]])
```

File: numpy/core/tests/test_einsum.py [hunk 2]
```diff
@@ -605,6 +605,11 @@ def test_einsum_misc(self):
                      [[[1,  3], [3,  9], [5, 15], [7, 21]],
                      [[8, 16], [16, 32], [24, 48], [32, 64]]])
 
+        # Ensure explicitly setting out=None does not cause an error
+        # see issue #15776 and issue #15256
```
Comment (eric-wieser) [inline numpy/core/tests/test_einsum.py]: Future-proofing in case we ever switch issue tracker again:
```suggestion
        # see gh-15776 and gh-15256
```

File: numpy/core/tests/test_einsum.py (hunk 1 above)
Comment (sgasse) [reply numpy/core/tests/test_einsum.py]: Ah OK! I though from the context that test results should go on a new line. But sure, from the line length, it can be on one line.

File: numpy/core/tests/test_einsum.py (hunk 2 above)
Comment (sgasse) [reply numpy/core/tests/test_einsum.py]: Alright thank you, will also update that :+1:

**Comment by sgasse** [comment]
Thank you very much for reviewing @eric-wieser!
Could I ask you two beginner questions?
- Using e.g. `gh-15776` should only be done in the code comments, right? In the commit message, I still use the `#` so that github picks up that I reference this issue, correct? Or would it also understand the `gh-15776` in the commit message?
- I worked in your suggestions, squashed the new commit into the old and force-pushed. Is this also the idea here in the numpy project on github? (I normally work with gerrit and patchsets)

File: numpy/core/tests/test_einsum.py (hunk 1 above)
Comment (eric-wieser) [reply numpy/core/tests/test_einsum.py]: If you wanted to do that, the wrapping would be
```suggestion
        assert_equal(np.einsum('i,j', [1], [2], out=None),
                     [[2]])
```
The way you wrapped it made it look like an argument to einsum.

But in general, test results on the same line are fine for short tests.

**Comment by eric-wieser** [comment]
> * Using e.g. `gh-15776` should only be done in the code comments, right? In the commit message, I still use the `#` so that github picks up that I reference this issue, correct? Or would it also understand the `gh-15776` in the commit message?

You can use either  - github understands both: gh-15776, £15776

> I worked in your suggestions, squashed the new commit into the old and force-pushed. Is this also the idea here in the numpy project on github? (I normally work with gerrit and patchsets)

It varies by the complexity of the change, but for a small one like this that's perfect, thanks!

For larger changes, having the iterative history makes it a little easier to re-review.

**Review by eric-wieser — APPROVED** [review]

**Comment by sgasse** [comment]
> You can use either - github understands both: [gh-15776](https://github.com/numpy/numpy/issues/15776), £15776

OK good to know, then I'll go with the `gh-` way in the future.

> It varies by the complexity of the change, but for a small one like this that's perfect, thanks!
> 
> For larger changes, having the iterative history makes it a little easier to re-review.

Alright, thank you for the info!

I guess the failing `azure-pipeline ComprehensiveTests Linux_PyPy3` is related to gh-15796?

**Comment by eric-wieser** [comment]
Yeah, that failure can be ignored.

**Comment by seberg** [comment]
Thanks all!
